src/routes/admin.routes.js

```diff
@@ -0,0 +1,42 @@
+const express = require('express')
+const router = new express.Router()
+// const { app } = require('./routes')
+
+const User = require('../db/userModel')
+const auth = require('../middlewares/auth')
+const isAdmin = require('../middlewares/admin')
+
+router.get('/users', auth, isAdmin, async (req, res) => {
+    try {
+        const users = await User.find({})
+        if (!users) return res.sendStatus(400)
+        res.render('adminView', {
+            users
+        })
+    } catch (error) {
+        res.sendStatus(500)
+    }
+})
+
+router.post('/users/update', auth, isAdmin, async (req, res) => {
+    try {
+        const { id, role } = req.body
+        const user = await User.findOne({ _id: id })
+
+        if (!user) {
+            return res.sendStatus(400)
```

Comment:
The catch block logs the error to the console, which is good for debugging but consider also informing the admin of what went wrong through the response.